solutions/USACO dec19-milkvisits-silver.mdx

```diff
@@ -0,0 +1,224 @@
+---
+id: usaco-968
+source: USACO Silver 2019 December
+title: Milk Visits
+author: Qi Wang
+---
+
+[Official Analysis (Offline)](http://www.usaco.org/current/data/sol_milkvisits_silver_dec19.html)
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+//Official Solution
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef vector<int> vi; 
+ 
+#define FOR(i,a,b) for (int i = (a); i < (b); ++i)
+#define F0R(i,a) FOR(i,0,a)
+#define ROF(i,a,b) for (int i = (b)-1; i >= (a); --i)
+#define R0F(i,a) ROF(i,0,a)
+#define trav(a,x) for (auto& a: x)
+ 
+#define pb push_back
+ 
+void setIO(string name) {
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen((name+".in").c_str(),"r",stdin);
+	freopen((name+".out").c_str(),"w",stdout);
+}
+
+const int MX = 100005;
+
+int N,M,comp[MX],num;
+char col[MX];
+vi adj[MX];
+ 
+void dfs(int x) {
+	if (comp[x]) return;
+	comp[x] = num;
+	trav(t,adj[x]) if (col[t] == col[x]) dfs(t);
+}
+ 
+int main() {
+	setIO("milkvisits"); cin >> N >> M;
+	string s; cin >> s;
+	FOR(i,1,N+1) col[i] = s[i-1];
```

Comment:
Just personal preference: can we change `col` to a boolean array? The reason this solutions works relies on the fact that there are only 2 cow breeds, so I think it would be nice to explicitly feature that constraint in code as well.

Reply:
yea sure I can do that, should I mention that this is different from the official one then? see comment below

Reply:
There's already a link to the official soln, so we can choose to not copy it (albeit it doesn't hurt if we do). Also, I don't think anything explicitly says that it is a copy paste of the official implementation, so we can change it if we'd like.